proto/stride/icqoracle/icqoracle.proto

```diff
@@ -0,0 +1,67 @@
+syntax = "proto3";
+package stride.icqoracle;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/Stride-Labs/stride/v24/x/icqoracle/types";
+
+// TokenPrice stores latest price data for a token
+message TokenPrice {
+  // Token denom on Stride
+  string base_denom = 1;
+  // Quote denom on Stride
+  string quote_denom = 2;
+  // Token denom on Osmosis
+  string osmosis_base_denom = 3;
+  // Quote denom on Osmosis
+  string osmosis_quote_denom = 4;
+  // Pool ID on Osmosis
+  string osmosis_pool_id = 5;
```

Comment:
@sampocs sampocs Dec 17, 2024

i wonder if it's worth adding a human readable name to this too? or maybe we can just enrich it with the full denom trace during the RPC query

Thinking being that if we do this with all our host zone tokens, then we'll have a bunch of IBC denom's and it gets pretty clunky to try and manually inspect things

@assafmo assafmo Dec 17, 2024

yeah i don't like the fields here too, i'll try to think of a better way to represent this

Reply:
i think internally we should just keep it as is cause there's already so much on the struct. But I think we could add a query that figures out the denom traces